internal/vault/credentials/jwt.go

```diff
@@ -95,13 +97,27 @@ func (l *JWTCredentialProvider) GetCreds(ctx context.Context, client ctrlclient.
 	}
 
 	var err error
-	l.tokenSecret, err = getSecret(ctx, client, l.providerNamespace, l.authObj.Spec.JWT.SecretKeyRef.Name)
+	key := ctrlclient.ObjectKey{
+		Namespace: l.providerNamespace,
+		Name:      l.authObj.Spec.JWT.SecretRef,
+	}
+	l.tokenSecret, err = getSecret(ctx, client, key)
 	if err != nil {
 		return nil, err
 	}
-
+	jwtData := l.tokenSecret.Data[ProviderSecretKeyJWT]
```

Comment:
nit: couldn't you do the get + existence check e.g: `jwtData, ok := l.tokenSecret.Data[ProviderSecretKeyJWT]`